ach/builder.py

```diff
@@ -79,35 +79,38 @@ def add_batch(self, std_ent_cls_code, batch_entries=None,
             company_name=(company_name or self.settings['company_name'])[:16],
         )
 
-        entries = []
+        entries, failed_entries = [], []
         entry_counter = 1
 
         for record in batch_entries:
+            try:
+                entry = EntryDetail(
+                    std_ent_cls_code=std_ent_cls_code,
+                    id_number=record.get('id_number', ''),
+                )
 
-            entry = EntryDetail(
-                std_ent_cls_code=std_ent_cls_code,
-                id_number=record.get('id_number', ''),
-            )
-
-            entry.transaction_code = record.get('type')
-            entry.recv_dfi_id = record.get('routing_number')
+                entry.transaction_code = record.get('type')
+                entry.recv_dfi_id = record.get('routing_number')
 
-            if len(record['routing_number']) < 9:
-                entry.calc_check_digit()
-            else:
-                entry.check_digit = record['routing_number'][8]
+                if len(record['routing_number']) < 9:
+                    entry.calc_check_digit()
+                else:
+                    entry.check_digit = record['routing_number'][8]
 
-            entry.dfi_acnt_num = record['account_number']
-            entry.amount = int(round(float(record['amount']) * 100))
-            entry.ind_name = record['name'].upper()[:22]
-            entry.trace_num = self.settings['immediate_dest'][:8] \
-                + entry.validate_numeric_field(entry_counter, 7)
+                entry.dfi_acnt_num = record['account_number']
+                entry.amount = int(round(float(record['amount']) * 100))
+                entry.ind_name = record['name'].upper()[:22]
+                entry.trace_num = self.settings['immediate_dest'][:8] \
+                    + entry.validate_numeric_field(entry_counter, 7)
 
-            entries.append((entry, record.get('addenda', [])))
-            entry_counter += 1
+                entries.append((entry, record.get('addenda', [])))
+                entry_counter += 1
+            except:
```

Comment:
Can we return what type of failure it was? 

Swallowing the error on a bunch of lines will help but, I believe we can do a better job by informing the user what the errors are or where they happened.

Reply:
Sure how do you think we should return the error? Should I add the error message in the entry? Or have failed_entries a tuple of entry dict and error message?

Reply:
I like the tuple idea.

Reply:
Tuple is ok for now but we could return an either type in the future.

Reply:
Hmm, Felippe, looking closer at your comment you're totally right. Can we log it with some kind of unique identifier to find that detail if not obvious from the data itself returned to the caller? This is a library so not sure what the convention is in python for logging in libraries, thoughts?